vips/image_golden_test.go

```diff
@@ -0,0 +1,221 @@
+package vips
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestImage_Resize_Downscale(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit.jpg", nil, func(img *ImageRef) error {
+		return img.Resize(0.9, KernelLanczos3)
+	})
+}
+
+func TestImage_Resize_Upscale(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit.jpg", nil, func(img *ImageRef) error {
+		return img.Resize(1.1, KernelLanczos3)
+	})
+}
+
+func TestImage_Resize_Downscale_Alpha(t *testing.T) {
+	goldenTest(t, resources+"png-8bit+alpha.png", nil, func(img *ImageRef) error {
+		return img.Resize(0.9, KernelLanczos3)
+	})
+}
+
+func TestImage_Resize_Upscale_Alpha(t *testing.T) {
+	goldenTest(t, resources+"png-8bit+alpha.png", nil, func(img *ImageRef) error {
+		return img.Resize(1.1, KernelLanczos3)
+	})
+}
+
+func TestImage_OptimizeICCProfile_CMYK(t *testing.T) {
+	goldenTest(t, resources+"jpg-32bit-cmyk-icc-swop.jpg", nil, func(img *ImageRef) error {
+		return img.OptimizeICCProfile()
+	})
+}
+
+func TestImage_OptimizeICCProfile_RGB_No_Profile(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit.jpg", nil, func(img *ImageRef) error {
+		return img.OptimizeICCProfile()
+	})
+}
+
+func TestImage_OptimizeICCProfile_RGB_Embedded(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit-icc-adobe-rgb.jpg", nil, func(img *ImageRef) error {
+		return img.OptimizeICCProfile()
+	})
+}
+
+func TestImage_OptimizeICCProfile_Grey(t *testing.T) {
+	goldenTest(t, resources+"jpg-8bit-gray-scale-with-icc-profile.jpg", nil, func(img *ImageRef) error {
+		return img.OptimizeICCProfile()
+	})
+}
+
+func TestImage_RemoveICCProfile(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit-icc-smpte.jpg", nil, func(img *ImageRef) error {
+		return img.RemoveICCProfile()
+	})
+}
+
+func TestImage_RemoveMetadata(t *testing.T) {
```

Comment:
maybe check that metadata was removed

Reply:
Again, golden test, check once on the golden image